frontend/src/features/statistics/StatsContext.tsx

```diff
@@ -52,23 +53,28 @@ export const StatsLazyProvider = ({
    * Initialize the stats if they are empty or refresh them if they are
    * outdated.
    */
-  const getStats = useCallback(() => {
-    const currentTime = Date.now();
+  const getStats = useCallback(
+    (poll: string | undefined) => {
+      const currentTime = Date.now();
 
-    if (loading.current) {
-      return stats;
-    }
+      if (loading.current) {
+        return stats;
+      }
```

Comment:
In some cases, this means `getStats()` could return stats about the wrong poll, because a concurrent request is running. This is probably not a problem for now, as there is a single one active poll, but it would be worth adding a FIXME.

```suggestion
      // FIXME: this function may return stats about the wrong poll here.
      // Maybe `useStats()` should always depend on `useCurrentPoll()` to simplify this logic.
      if (loading.current) {
        return stats;
      }
```

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1830/commits/094f0306861a316dcc50357ac463bc810dca94af